### PR TITLE
feat(regular_expression): Support RegExp Modifiers

### LINKED ISSUE #6354: regular_expression: proposal-regexp-modifiers stage 4
https://github.com/tc39/proposal-regexp-modifiers

https://github.com/oxc-project/oxc/blob/4fd89e8eeda9314cc4e74cd71607e4a7ae4534e5/tasks/coverage/src/test262/mod.rs#L110-L111

### PR DESCRIPTION
**Pull request by leaysgur**
Fixes #6354

### CODE REVIEW

File: crates/oxc_regular_expression/src/parser/parser_impl.rs [hunk 1]
```diff
@@ -1525,30 +1533,137 @@ impl<'a> Parser<'a> {
...
+            ));
         }
 
-        Ok(None)
+        // NOTE: Spec is not yet fixed and merged, so these may change:
+        // - Both empty: `(?-:.)`
+        // - The second is empty: `(?i-:.)`
+        // All of V8, Babel and TS emit error for both empty case, but only Babel emits error for the second empty case.
+        // For now, we follow Babel's behavior.
+        // https://github.com/tc39/ecma262/pull/3221#pullrequestreview-2341169958
```
[the remark this answers is not in the document]
Comment (Boshen) [reply crates/oxc_regular_expression/src/parser/parser_impl.rs:1647]: If there are no such cases in test262, then we should not throw it.
We aim to conform with test262, not other implementations.

[the remark this answers is not in the document]
Comment (leaysgur) [reply crates/oxc_regular_expression/src/parser/parser_impl.rs]: The latter case seems allowed.

https://github.com/tc39/test262/blob/954d6370f99b22fbce174562f6718dd69afd3094/test/built-ins/RegExp/regexp-modifiers/syntax/valid/add-and-remove-modifiers-can-have-empty-remove-modifiers.js#L7

But the former is not mentioned, so leave it as is for now.

**Comment by Boshen** [comment]
### Merge activity

* **Oct 10, 1:01 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 10, 1:01 AM EDT**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

**Comment by leaysgur** [comment]
Not merged properly due to bad timing? 🤔

**Pull request merged**